sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/DataFeedSourcesTests.cs

```diff
@@ -72,10 +88,84 @@ public async Task DataFeedSourceSendsSecretDuringCreation(DataFeedSource dataSou
 
             MockRequest request = mockTransport.Requests.First();
             string content = ReadContent(request);
+            string expectedSubstring = $"\"{secretPropertyName}\":\"secret\"";
 
             Assert.That(content, Contains.Substring(expectedSubstring));
         }
 
+        [Test]
+        [TestCaseSource(nameof(UpdateDataSourceTestCases))]
+        public async Task DataFeedSourceSendsSecretDuringUpdate(DataFeedSource dataSource, string secretPropertyName)
+        {
+            MockResponse updateResponse = new MockResponse(200);
+            updateResponse.SetContent(DataFeedResponseContent);
+
+            MockTransport mockTransport = new MockTransport(updateResponse);
+            MetricsAdvisorAdministrationClient adminClient = CreateInstrumentedAdministrationClient(mockTransport);
+
+            UpdateSecret(dataSource, secretPropertyName);
+
+            DataFeed dataFeed = new DataFeed()
+            {
+                Id = FakeGuid,
+                Name = "name",
+                DataSource = dataSource,
+                Granularity = new DataFeedGranularity(DataFeedGranularityType.Daily),
+                Schema = new DataFeedSchema(),
+                IngestionSettings = new DataFeedIngestionSettings() { IngestionStartTime = DateTimeOffset.UtcNow }
+            };
+
+            await adminClient.UpdateDataFeedAsync(dataFeed);
+
+            MockRequest request = mockTransport.Requests.First();
+            string content = ReadContent(request);
+
+            Assert.That(content, Contains.Substring($"\"{secretPropertyName}\":\"new_secret\""));
+        }
+
+        private void UpdateSecret(DataFeedSource dataSource, string secretPropertyName)
+        {
+            switch (dataSource)
+            {
+                case AzureApplicationInsightsDataFeedSource d:
+                    d.UpdateApiKey("new_secret");
+                    break;
+                case AzureBlobDataFeedSource d:
+                    d.UpdateConnectionString("new_secret");
+                    break;
+                case AzureCosmosDbDataFeedSource d:
+                    d.UpdateConnectionString("new_secret");
+                    break;
+                case AzureDataExplorerDataFeedSource d:
+                    d.UpdateConnectionString("new_secret");
+                    break;
+                case AzureDataLakeStorageGen2DataFeedSource d:
+                    d.UpdateAccountKey("new_secret");
+                    break;
+                case AzureTableDataFeedSource d:
+                    d.UpdateConnectionString("new_secret");
+                    break;
+                case InfluxDbDataFeedSource d:
+                    if (secretPropertyName == "connectionString") d.UpdateConnectionString("new_secret");
+                    else d.UpdatePassword("new_secret");
+                    break;
```

Comment:
`InfluxDbDataFeedSource` has two secrets, so we use the `secretPropertyName` parameter to know which one to update.